dwave/preprocessing/include/dwave/presolve.h

```diff
@@ -511,7 +621,9 @@ void Presolver<bias_type, index_type, assignment_type>::apply() {
         changes |= technique_check_for_nan();
         // *-- remove single variable constraints
         changes |= technique_remove_single_variable_constraints();
-        // *-- tighten bounds based on vartype
+        // *-- domain propagation
```

Comment:
Is this the right order or we want to do the domain propagation after tightening the bounds?

Reply:
I think I would first tighten the bounds and then domain propagate. At the beginning of paragraph 3.2 of the paper you followed (bottom of page 7), they assume that the bounds are already tightened, so it makes sense to do that first.

Reply:
Right! Thanks Alex. Done.